operator/pkg/resources/statefulset_update.go

```diff
@@ -585,19 +666,31 @@ func (r *StatefulSetResource) shouldUpdate(
 	if err != nil || patchResult.IsEmpty() {
 		return false, err
 	}
+	log.Info("Detected diff", "patchResult", string(patchResult.Patch))
 	return true, nil
 }
 
 func (r *StatefulSetResource) getRestartingStatus() bool {
 	return r.pandaCluster.Status.IsRestarting()
 }
 
-func (r *StatefulSetResource) updateRestartingStatus(
-	ctx context.Context, restarting bool,
-) error {
-	if !reflect.DeepEqual(restarting, r.getRestartingStatus()) {
-		r.pandaCluster.Status.SetRestarting(restarting)
+func (r *StatefulSetResource) updateRestartingStatus(ctx context.Context, restarting bool) error {
+	npStatus := r.getNodePoolStatus()
+
+	if !reflect.DeepEqual(restarting, npStatus.Restarting) {
```

Comment:
I realize that this is pre-existing but why are we using DeepEqual to compare booleans?

Reply:
i have no idea. Kept things as-is rather than refactoring more than needed